### ISSUE TITLE
Undo with pencil on a stack behaves strangely

### ISSUE THREAD

**Issue by ctrueden** [imagej/imagej1#47]
Ran into this bug with @milkyklim today:

* Create a new image with:
    ```javascript
    newImage("Untitled", "8-bit ramp", 64, 64, 2);
    ```
* Select the pencil tool and draw a squiggle.
* Undo.
* Draw another squiggle. The first squiggle reappears.

It works as expected with a 2D image.

**Comment by imagejan** [comment]
Strange, but I cannot reproduce in my up-to-date Fiji on Windows 7.
<details>

```
  java.version: 1.8.0_144
  java.vendor: Oracle Corporation
  os.name: Windows 7
  os.version: 6.1
  os.arch: amd64
  file.separator: \
  path.separator: ;
  line.separator: <cr><lf>

  user.name: eglijan
  user.home: C:\Users\eglijan
  user.dir: C:\Utilities\debug\Fiji.app
  user.country: US
  file.encoding: Cp1252
  java.home: C:\Program Files\Java\jdk1.8.0_144\jre
  java.class.path: C:\UTILIT~1\debug\Fiji.app/jars/imagej-launcher-4.0.5.jar
  java.ext.dirs: C:\Program Files\Java\jdk1.8.0_144\jre\lib\ext;C:\Windows\Sun\Java\lib\ext
  java.io.tmpdir: C:\Users\eglijan\AppData\Local\Temp\

  IJ.getVersion: 2.0.0-rc-61/1.51n
  IJ.getFullVersion: 1.51n99
  IJ.isJava17: true
  IJ.isJava18: true
  IJ.isLinux: false
  IJ.isMacintosh: false
  IJ.isMacOSX: false
  IJ.isWindows: true
  IJ.is64Bit: true

  IJ.getDir("imagej"): C:\UTILIT~1\debug\Fiji.app\
  IJ.getDir("home"): C:\Users\eglijan\
  IJ.getDir("plugins"): C:\UTILIT~1\debug\Fiji.app\plugins\
  IJ.getDir("macros"): C:\UTILIT~1\debug\Fiji.app\macros\
  IJ.getDir("luts"): C:\UTILIT~1\debug\Fiji.app\luts\
  IJ.getDir("current"): E:\eglijan\projects\ferejaro\
  IJ.getDir("temp"): C:\Users\eglijan\AppData\Local\Temp\
  IJ.getDir("default"): E:\eglijan\projects\ferejaro\
  IJ.getDir("image"): null

  Menus.getPlugInsPath: C:\UTILIT~1\debug\Fiji.app\plugins\
  Menus.getMacrosPath: C:\UTILIT~1\debug\Fiji.app\macros\
  Prefs.getImageJDir: C:\UTILIT~1\debug\Fiji.app\
  Prefs.getThreads: 16 (16 cores)
  Prefs.open100Percent: false
  Prefs.blackBackground: true
  Prefs.useJFileChooser: false
  Prefs.weightedColor: false
  Prefs.blackCanvas: false
  Prefs.pointAutoMeasure: false
  Prefs.pointAutoNextSlice: false
  Prefs.requireControlKey: false
  Prefs.useInvertingLut: false
  Prefs.antialiasedTools: true
  Prefs.useInvertingLut: false
  Prefs.intelByteOrder: false
  Prefs.doubleBuffer: true
  Prefs.noPointLabels: false
  Prefs.disableUndo: false
  Prefs dir: C:\Users\eglijan\.imagej
  Current dir: E:\eglijan\projects\ferejaro\
  Sample images dir: http://wsr.imagej.net/images/
  Memory in use: 7633MB of 98139MB (7%)
  Screen size: 2560x1600
  Max window bounds: x=0,y=0,width=2560,height=1560
  Monitor1: x=0,y=0,width=2560,height=1600
  Monitor2: x=-1200,y=0,width=1200,height=1600

All Java Properties
  jnlp.jogamp.tmp.cache.root: jln3356803010721386009
  ij.dir: C:\UTILIT~1\debug\Fiji.app
  java.runtime.name: Java(TM) SE Runtime Environment
  fiji.dir: C:\UTILIT~1\debug\Fiji.app
  imagej.dir: C:\UTILIT~1\debug\Fiji.app
  sun.boot.library.path: C:\Program Files\Java\jdk1.8.0_144\jre\bin
  java.vm.version: 25.144-b01
  user.country.format: CH
  fiji.defaultLibPath: bin/server/jvm.dll
  java.vm.vendor: Oracle Corporation
  java.vendor.url: http://java.oracle.com/
  path.separator: ;
  java.vm.name: Java HotSpot(TM) 64-Bit Server VM
  file.encoding.pkg: sun.io
  user.country: US
  user.script: 
  sun.os.patch.level: Service Pack 1
  java.vm.specification.name: Java Virtual Machine Specification
  user.dir: C:\Utilities\debug\Fiji.app
  java.runtime.version: 1.8.0_144-b01
  scijava.context.strict: false
  java.awt.graphicsenv: sun.awt.Win32GraphicsEnvironment
  java.endorsed.dirs: C:\Program Files\Java\jdk1.8.0_144\jre\lib\endorsed
  os.arch: amd64
  java.io.tmpdir: C:\Users\eglijan\AppData\Local\Temp\
  line.separator: 

  java.vm.specification.vendor: Oracle Corporation
  user.variant: 
  os.name: Windows 7
  sun.jnu.encoding: Cp1252
  java.library.path: C:\UTILIT~1\debug\Fiji.app/lib/win64;C:\UTILIT~1\debug\Fiji.app/mm/win64;C:\Users\eglijan\wekafiles\native
  sun.awt.enableExtraMouseButtons: true
  java.specification.name: Java Platform API Specification
  java.class.version: 52.0
  sun.management.compiler: HotSpot 64-Bit Tiered Compilers
  os.version: 6.1
  user.home: C:\Users\eglijan
  user.timezone: Europe/Berlin
  java.awt.printerjob: sun.awt.windows.WPrinterJob
  file.encoding: Cp1252
  java.specification.version: 1.8
  java.net.useSystemProxies: true
  plugins.dir: C:\UTILIT~1\debug\Fiji.app
  java.class.path: C:\UTILIT~1\debug\Fiji.app/jars/imagej-launcher-4.0.5.jar
  user.name: eglijan
  java.vm.specification.version: 1.8
  java.home: C:\Program Files\Java\jdk1.8.0_144\jre
  sun.arch.data.model: 64
  WEKA_HOME: C:\Users\eglijan\wekafiles
  user.language: en
  java.specification.vendor: Oracle Corporation
  user.language.format: de
  awt.toolkit: sun.awt.windows.WToolkit
  java.vm.info: mixed mode
  java.version: 1.8.0_144
  java.ext.dirs: C:\Program Files\Java\jdk1.8.0_144\jre\lib\ext;C:\Windows\Sun\Java\lib\ext
  sun.boot.class.path: C:\Program Files\Java\jdk1.8.0_144\jre\lib\resources.jar;C:\Program Files\Java\jdk1.8.0_144\jre\lib\rt.jar;C:\Program Files\Java\jdk1.8.0_144\jre\lib\sunrsasign.jar;C:\Program Files\Java\jdk1.8.0_144\jre\lib\jsse.jar;C:\Program Files\Java\jdk1.8.0_144\jre\lib\jce.jar;C:\Program Files\Java\jdk1.8.0_144\jre\lib\charsets.jar;C:\Program Files\Java\jdk1.8.0_144\jre\lib\jfr.jar;C:\Program Files\Java\jdk1.8.0_144\jre\classes
  java.vendor: Oracle Corporation
  ij.executable: C:\Utilities\debug\Fiji.app\ImageJ-win64.exe
  python.cachedir.skip: true
  file.separator: \
  java.vendor.url.bug: http://bugreport.sun.com/bugreport/
  sun.io.unicode.encoding: UnicodeLittle
  sun.cpu.endian: little
  sun.desktop: windows
  fiji.executable: C:\Utilities\debug\Fiji.app\ImageJ-win64.exe
  sun.cpu.isalist: amd64
```
</details>

**Comment by rasband** [comment]
I am unable to reproduce this problem with ImageJ and Fiji on macOS.

**Comment by ctrueden** [comment]
I tried several times to reproduce under various conditions. I was able to reproduce every time with Fiji 2.0.0-rc-43/1.51h + Java 1.6.0_24 on Windows 10, and with Fiji 2.0.0-rc-61/1.51n + Java 1.8.0_144 on macOS High Sierra. But I could never reproduce it with 1.51s, nor with plain ImageJ 1.51n launched via `java -jar`.

Closing for now. I will reopen if I run into the problem with the latest version of ImageJ.
